### PR TITLE
GEOMESA-3170 Allow for specifying different shard numbers for Z2 and …

### PR DESCRIPTION
**Pull request by aheyne**
…Z3 indices.

- Adds config keys geomesa.z2.splits and geomesa.z3.splits

Signed-off-by: Austin Heyne <aheyne@ccri.com>

### CODE REVIEW

File: geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/SimpleFeatureTypes.scala [hunk 1]
```diff
@@ -64,6 +63,11 @@ object SimpleFeatureTypes {
     val UpdateBackupMetadata  = "schema.update.backup.metadata"
     val UpdateRenameTables    = "schema.update.rename.tables"
 
+    @deprecated("replaced with IndexZ2Shards and IndexZ3Shards")
```
Comment (elahrvivaz) [inline geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/SimpleFeatureTypes.scala:66]: Could we leave z.splits as undeprecated, and just make the others take precedence if present? that would simplify config in the general case, and you've already accounted for the others having higher priority.

Comment (aheyne) [reply geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/SimpleFeatureTypes.scala:66]: Sure, do you want the same done in the schema builder?

**Comment by elahrvivaz** [comment]
lgtm, I'd just like to keep the original config undeprecated.

**Comment by jnh5y** [comment]
Also, is it worth updating the Git commit to say "Deprecates" instead of "Depricates"?;)

File: geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/Conversions.scala [hunk 2]
```diff
@@ -303,8 +303,13 @@ object RichSimpleFeatureType extends Conversions {
     def getUserDataPrefixes: Seq[String] =
       Seq(GeomesaPrefix) ++ userData[String](UserDataPrefix).map(_.split(",")).getOrElse(Array.empty)
 
-    def setZShards(splits: Int): Unit = sft.getUserData.put(IndexZShards, splits.toString)
-    def getZShards: Int = userData(IndexZShards).map(int).getOrElse(4)
+    def setZ2Shards(splits: Int): Unit = sft.getUserData.put(IndexZ2Shards, splits.toString)
```
Comment (elahrvivaz) [inline geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/Conversions.scala:306]:
```suggestion
    def setZShards(splits: Int): Unit = sft.getUserData.put(IndexZShards, splits.toString)
    @deprecated("use z2/z3 specific methods")
    def getZShards: Int = userData(IndexZShards).map(int).getOrElse(4)
    def setZ2Shards(splits: Int): Unit = sft.getUserData.put(IndexZ2Shards, splits.toString)
```

**Pull request merged**